File: GrowthWare_VB/GrowthWareFramework/DataAccessLayer/Interfaces/IGroups.vb
Imports GrowthWare.Framework.DataAccessLayer.Interfaces.Base
Imports GrowthWare.Framework.ModelObjects

Namespace DataAccessLayer
	Public Interface IGroups
		Inherits IDBInteraction

		Property Profile() As MGroupProfile
		Property GroupRolesProfile() As MGroupRoles

		Function GetGroupRoles() As DataTable
		Function UpdateGroupRoles() As Boolean

		Function GetGroupsByBU() As DataTable
		Function AddGroup() As Boolean
		Function GetProfileData() As DataRow
		Function DeleteGroup() As Boolean
		Function UpdateGroup() As Boolean
	End Interface
End Namespace

File: Net_Core/DataAccess/Interfaces/IGroups.cs
using GrowthWare.DataAccess.Interfaces.Base;
using GrowthWare.Framework.Models;
using System.Data;
using System.Threading.Tasks;

namespace GrowthWare.DataAccess.Interfaces;

public interface IGroups : IDBInteraction
{

    /// <summary>
    /// GroupProfile
    /// </summary>
    MGroupProfile Profile { get; set; }

    /// <summary>
    /// GroupRoles
    /// </summary>
    MGroupRoles GroupRolesProfile { get; set; }

    /// <summary>
    /// Returns a DataTable of Group roles
    /// </summary>
    /// <returns>DataTable</returns>
    Task<DataTable> GroupRoles();

    /// <summary>
    /// Updates the Groups roles
    /// </summary>
    /// <returns>bool</returns>
    Task<bool> UpdateGroupRoles();

    /// <summary>
    /// Get's all of the groups for a given Security Entity
    /// </summary>
    /// <returns>DataTable</returns>
    Task<DataTable> GroupsBySecurityEntity(int securityEntityId);

    /// <summary>
    /// Returns a data row necessary to populate MGroupProfile
    /// </summary>
    /// <returns>DataRow</returns>
    Task<DataRow> ProfileData(int securityEntityId);

    /// <summary>
    /// Deletes a group in a given Security Entity
    /// </summary>
    /// <returns>bool</returns>
    Task<bool> DeleteGroup();

    /// <summary>
    /// Saves this instance.
    /// </summary>
    Task<int> Save();
}